api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -473,7 +494,11 @@ public NestedField asRequired() {
       if (!isOptional) {
         return this;
       }
-      return new NestedField(false, id, name, type, doc);
+      return new NestedField(false, id, name, type, doc, initialDefault, writeDefault);
+    }
+
+    public NestedField updateWriteDefault(Object newWriteDefault) {
```

Comment:
I don't think there are other methods like this. Is it worth adding a new pattern for just this? Why not add `updateDoc`? I think I'd rather copy where needed.